Added functionality for having multiple packets open at the same time.

realDragonium opened the pull request:
Hey,

I was looking for something like this, but I thought having multiple packets open at the same time works a bit better when you want to compare packets with each other.  

And my IDE found it a good idea if some dependencies got added/were updated. :sweat_smile:

realDragonium committed:
Added functionality for having multiple packages open at the same time.

wvffle commented on src/App.vue:
```diff
@@ -381,7 +380,7 @@
         },
         listening: false,
-        id_open: null,
+        open_ids: [],
```
Use a `new Set()` rather than an array

wvffle commented on src/App.vue:
```diff
@@ -252,12 +252,11 @@
+          return this.open_ids.splice(index, 1);
         }
-
-        window.packet = packet
```
That was for quick debugging in console. I'd rather not resign from this behavior. Having an ability to manipulate over opened packet (packets in case of this PR) in console is a must have in my case.

realDragonium replied:
Ah, so that's what it was for. I have no problem with it so I will put it back. How can you use this?

wvffle reviewed (changes requested):
Two minor issues but thanks for PR!

It still has a couple of bugs as I left it in the middle of rewriting it. I have no free time right now to fix them so if you're willing to fix them, it'd be awesome!

realDragonium commented:
Let me know what those bugs are and I will see what I can do about them!

I have also been working on making it work for online-mode servers. Did this also included the scope of your project?

wvffle commented:
> Let me know what those bugs are and I will see what I can do about them!

I've highlighted them in the review

> I have also been working on making it work for online-mode servers. Did this also included the scope of your project?

Yup, right now localhost 1.12.2 is hardcoded as far as I remember. You should be able to change the server in the client settings

realDragonium committed:
Main page of the website now also loads settings.
There is now an option for toggling or you want to open multiple packets at the same time.
Rename `open_ids` to `open_packets`. Also have tried to make a `Set` of it, but it didnt open them when you clicked on them.

wvffle commented:
idk if it's still there but you can trace `server.host` variable in index.js of https://github.com/wvffle/minecraft-packet-debugger/commit/8153464afa821dd5167c399e5e4ac5a43bd30b71. There's some stuff that's responsible for changing the server

realDragonium commented:
> Use a new Set() rather than an array

It doesnt work anymore when its a `Set` so it will have to stay as an array for now.

wvffle commented:
Lets stick with an array then. Can you expose open_packets to window and I'll merge it?

**Edit:** I've seen that you exported packet. That's ok for now

realDragonium committed:
exposed open_packets to window.packet

realDragonium commented:
```
window.packet = this.open_packets
```

like this? or should `window.open_packet` be better?
